pkgs/applications/science/math/eukleides/default.nix

```
@@ -1,26 +1,43 @@
 { lib, stdenv, fetchurl, bison, flex, texinfo, readline, texLive }:
 
-let
-  name    = "eukleides";
+lib.fix (eukleides: stdenv.mkDerivation rec {
+  pname = "eukleides";
   version = "1.5.4";
-in
-stdenv.mkDerivation {
-  name = "${name}-${version}";
 
   src = fetchurl {
-    url = "http://www.eukleides.org/files/${name}-${version}.tar.bz2";
+    url = "http://www.eukleides.org/files/${pname}-${version}.tar.bz2";
     sha256 = "0s8cyh75hdj89v6kpm3z24i48yzpkr8qf0cwxbs9ijxj1i38ki0q";
   };
 
-  buildInputs = [bison flex texinfo readline texLive];
+  nativeBuildInputs = [ bison flex ];
 
-  preConfigure = "sed -i 's/ginstall-info/install-info/g' doc/Makefile";
-  installPhase = "mkdir -p $out/bin ; make PREFIX=$out install";
+  buildInputs = [ texinfo readline texLive ];
+
+  preConfigure = ''
+    sed -i 's/ginstall-info/install-info/g' doc/Makefile
+    sed -i '/mktexlsr 2> \/dev\/null/d' Makefile
+  '';
```

```suggestion
    substituteInPlace Config \
      --replace '/usr/local' "$out" \
      --replace '$(SHARE_DIR)/doc' "$doc/share/doc" \
      --replace '$(SHARE_DIR)/man "$doc/share/man" \
      --replace '$(SHARE_DIR)/info "$doc/share/info" \
      --replace '$(SHARE_DIR)/texmf' "$tex" \
  '';
```
That should work better than overriding `installPhase` – it mimics what would happen if there were a `configure` script. Please double check that I have got that right, though!

(By the way, you can do the previous `sed`'s with `substituteInPlace` too.)

---

I'm having problems with escaping the `$(SHARE_DIR)` here 
```
/nix/store/6xss8bksa5hsjl981y9as0adrcyjfgkl-stdenv-linux/setup: line 86: SHARE_DIR: command not found
/nix/store/6xss8bksa5hsjl981y9as0adrcyjfgkl-stdenv-linux/setup: line 86: SHARE_DIR: command not found
substituteStream(): WARNING: pattern '$(SHARE_DIR)/man "$doc/share/man" \
  --replace /doc "$doc/share/doc" \
  --replace /info' doesn't match anything in file 'Config'
```
and it fails to substitute some, but not all, of the `--replace` arguments.
What is the magical sequence here? I tried `$` -> `\$` and `$` -> `$$`.

How would I replace the `/d` variant of `sed` by `substituteInPlace`? I don't see any `--delete` or similar option in the docs?

---

I am very sorry about that, I missed two single quotes (after `/man` and `/info`). That should fix the issue I hope.

> How would I replace the /d variant of sed by substituteInPlace? I don't see any --delete or similar option in the docs?

I would either replace `mktexlsr` with `true`, or with `# mktexlsr`. Indeed there is no delete option.

---

Haha, wow, I was staring at it for so long but I didn't see the missing quotes either. Downsides of being inside a literal string already, didn't have any extra syntax highlighting.